pkg/codegen/go/gen.go

```diff
@@ -444,6 +458,30 @@ func (pkg *pkgContext) typeStringImpl(t schema.Type, argsType bool) string {
 	panic(fmt.Errorf("unexpected type %T", t))
 }
 
+func (pkg *pkgContext) getImportPrefix(tok string, typ schema.Type) string {
+	var tokenRange hcl.Range
+	pkgName, module, _, _ := hcl2.DecomposeToken(tok, tokenRange)
+	// namespaceless invokes
+	if module == "" || strings.HasPrefix(module, "/") || strings.HasPrefix(module, "index/") {
+		module = pkgName
+	}
+	info, ok := pkg.pkg.Language["go"].(GoPackageInfo)
+	if !ok {
+		return module
+	}
+
+	if m, ok := info.ModuleToPackage[module]; ok {
+		module = m
+	}
+
+	imp := fmt.Sprintf("%s/%s", info.ImportBasePath, module)
+	if alias, ok := info.PackageImportAliases[imp]; ok {
+		imp = alias
+	}
+
+	return strings.Split(imp, "/")[0]
+}
```

Comment:
This was previously a part of a method of the `generator` before it was deleted in #7059. It now lies in the intertwined codepath that services both retrieving types for program gen and generating sdk types and input types. 

The closest thing seems to be `tokenToPackage` but this produces incorrect results. ie instead of `aws` it returns `/getamifilter`.